File: src/main/java/com/tw/pos/discount/SecondHalf.java
package com.tw.pos.discount;

import com.tw.pos.entity.CartItem;


public class SecondHalf implements Discount {

    private double price;

    @Override
    public double apply(CartItem cartItem, double price) {
        this.price = price;
        double sum = 0;
        for (int i = 1; i <= cartItem.getNumber(); i++){
            if (i % 2 == 0){
                sum += price / 2;
            } else {
                sum += price;
            }
        }
        return sum;
    }

    @Override
    public double getDiscountPrice(double originPrice) {
        return price;
    }
}

File: js/discount/secondHalf.js

var SecondHalf = function(){
};

SecondHalf.prototype.apply = function(cartItem, price){
    this.price = price;
    var sum = 0;
    for (var i = 1; i <= cartItem.number;i++){
        if (i % 2 == 0){
            sum += price / 2;
        }else {
            sum += price;
        }
    }
    return sum;
};

SecondHalf.prototype.getDiscountPrice = function(originPrice){
  return this.price;
};